ops/demo/main.tf

```diff
@@ -34,10 +34,8 @@ module "simple_report_api" {
     SPRING_LIQUIBASE_ENABLED                         = "true"
     SPRING_JPA_PROPERTIES_HIBERNATE_DEFAULT_SCHEMA   = "public"
     SPRING_DATASOURCE_URL                            = "@Microsoft.KeyVault(SecretUri=${data.azurerm_key_vault_secret.sr_db_jdbc.id})"
-    //    OKTA_OAUTH2_CLIENT_ID                            = "@Microsoft.KeyVault(SecretUri=${data.azurerm_key_vault_secret.okta_client_id.id})"
-    //    OKTA_OAUTH2_CLIENT_SECRET                        = "@Microsoft.KeyVault(SecretUri=${data.azurerm_key_vault_secret.okta_client_secret.id})"
-    OKTA_OAUTH2_ISSUER                    = "https://hhs-prime.okta.com/oauth2/default",
-    APPLICATIONINSIGHTS_CONNECTION_STRING = "InstrumentationKey=${data.azurerm_application_insights.app_insights.instrumentation_key};IngestionEndpoint=https://eastus-1.in.applicationinsights.azure.com/"
+    DATAHUB_API_KEY                                  = "@Microsoft.KeyVault(SecretUri=${data.azurerm_key_vault_secret.datahub_api_key.id})"
```

Comment:
So, if we want to inject this directly, it would be (I think)`SIMPLE-REPORT_DATA-HUB-API-KEY`. @TomNUSDS can confirm the property names--the key is that dashes remain dashes and dots/colons become underscores.

Reply:
I wasn't real confident about all the auto translations of naming conventions. I was following the OKTA auth data as an example of the best way to get key vault data into the java instances.

@benwarfield-usds What's the best way to verify that fully scoped variables in the Spring config file:
```
simple-report:
  data-hub:
    api-key: ${datahubApiKey:MISSING_PROD_DATAHUB_KEY}
```
So is it correct that the config.yaml's `simple-report.data-hub.api-key` maps to an environment variable  `SIMPLE_REPORT-DATA_HUB-API_KEY` and then that maps to the javascript tag `simpleReport.dataHub.apiKey`

We should discuss this at stand-up but is super confusing (and thus error prone) to use these automatic conversions everywhere.